base/src/main/java/io/spine/validate/Distinct.java

```diff
@@ -20,37 +20,49 @@
 
 package io.spine.validate;
 
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import io.spine.code.proto.Option;
 import io.spine.option.OptionsProto;
 
 import java.util.Optional;
 
 /**
  * An option that can be applied to {@code repeated} Protobuf fields to specify that values
- * represented by that {@code repeated} field don't contain duplicates.
+ * represented by that {@code repeated} field should not contain duplicates.
  *
  * @param <T>
- *         type fields that can be checked against this option
+ *         types of values that this option is applicable to
  */
-final class OnDuplicate<T> extends FieldValidatingOption<io.spine.option.OnDuplicate, T> {
+final class Distinct<T> extends FieldValidatingOption<Boolean, T> {
 
-    private OnDuplicate() {
-        super(OptionsProto.onDuplicate);
+    private Distinct() {
+        super(OptionsProto.distinct);
     }
 
     /**
      * Returns a new instance of this option.
      *
-     * @param <T> type of fields that can be checked against this option
+     * @param <T>
+     *         type of fields that can be checked against this option
      */
-    static <T> OnDuplicate<T> create() {
-        return new OnDuplicate<>();
+    static <T> Distinct<T> create() {
+        return new Distinct<>();
     }
 
     @Override
-    public Optional<io.spine.option.OnDuplicate> valueFrom(FieldValue<T> fieldValue) {
-        io.spine.option.OnDuplicate option = fieldValue.valueOf(optionExtension());
-        boolean isDefault = option.getNumber() == 0;
-        return isDefault ? Optional.empty() : Optional.of(option);
+    public Optional<Boolean> valueFrom(FieldValue<T> fieldValue) {
+        FieldDescriptor descriptor = fieldValue.declaration()
+                                               .descriptor();
+        Option distinct = Option.from(descriptor, optionExtension());
+
+        return distinct.isExplicitlySet()
```

Comment:
We have a repeated scenario  with this `isExplicitlySet()` thing:
 1. First we get a thing which is called `Option` 
     This is a part of the problem to me: this thing is a value of an option, not an option itself. The `Option` class does not have a default value and works differently to what we have in Protobuf (which has `Option.getValue()`).

 2. Then, we query if the thing `isExplicitlySet()`.
     In the absence of the notion of the default value in an option, the method looks strange.

3. Then, depending on the value returned by `isExplicitlySet()` we have return the value of the field wrapped into `Optional`, or `empty()`.

Why cannot `fieldValue.valueOf(optionExtension())` perform the checking and return corresponding `Optional`?

Reply:
@armiol, FYI.